hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -230,12 +233,25 @@ protected void commit(HoodieWriteMetadata<O> result, List<HoodieWriteStat> write
           serializeCommitMetadata(table.getMetaClient().getCommitMetadataSerDe(), metadata));
       LOG.info("Committed " + instantTime);
       result.setCommitMetadata(Option.of(metadata));
+      // update cols to Index as applicable
+      HoodieColStatsIndexUtils.updateColsToIndex(table, config, metadata,
+          (Functions.Function2<HoodieTableMetaClient, List<String>, Void>) (val1, val2) -> {
+            updateColumnsToIndexWithColStats(val1, val2);
+            return null;
+          });
     } catch (IOException e) {
       throw new HoodieCommitException("Failed to complete commit " + config.getBasePath() + " at time " + instantTime,
           e);
     }
   }
 
+  /**
+   * Updates the list of columns indexed with col stats index in Metadata table.
+   * @param metaClient instance of {@link HoodieTableMetaClient} of interest.
+   * @param columnsToIndex list of columns to index.
+   */
+  protected abstract void updateColumnsToIndexWithColStats(HoodieTableMetaClient metaClient, List<String> columnsToIndex);
```

Comment:
The concrete impl of this API and that of `BaseHoodieClient.updateColumnsToIndexWithColStats` is same (at least for Spark). Do we still need this API? By the way, isn't colstats supported for Flink? 

Regardless, I think engine specific methods for updating index definition does not fit well. Do we have such examples for updating table config?